uber_sante/controllers/booking_controller.py

```diff
@@ -29,25 +32,28 @@ def book():
     # TODO: connect the call to the scheduler class to try reserving the availability (line 51)
     # TODO: connect the call to the booking_service to create the booking in the Booking table (line 54)
 
-        appointment_id = request.args.get('appointment_id')
+        availability_id = request.args.get('availability_id')
         patient_id = request.args.get('patient_id')
 
-        if appointment_id is None:
-            return jsonify('No appointment specified'), 400
+        if availability_id is None:
+            return js.create_json(data=None, message="No appointment specified", return_code=js.ResponseReturnCode.CODE_400)
         if patient_id is None:
-            return jsonify('No patient specified'), 400
+            return js.create_json(data=None, message="No patient specified", return_code=js.ResponseReturnCode.CODE_400)
+
 
-        patient = get_from_cache(patient_id) # get the patient from cache
-        appointment = patient.cart[appointment_id] # somewhat of a placeholder for now, waiting for cart to be implemented
+        # TODO: Remove Dummy appointment below and use the implementation below
+        patient = get_from_cache(patient_id)  # get the patient from cache
+        appointment = patient.cart.get_appointment(availability_id)
+
+        result = Scheduler.get_instance().reserve_appointment(appointment)
 
-        result = True #scheduler.try_booking(appointment)   # make a call to the scheduler here
-                                                            # returns availability object
         if result:
-            True # booking_service.book(result, patient)
+            BookingService().write_booking(appointment)
+            return js.create_json(data=None, message="Appointment successfully booked", return_code=js.ResponseReturnCode.CODE_200)
```

Comment:
return the appointment object to the front end (in the data=None) incase we want to display it

Reply:
You also need to add the "remove from cart" part. Just call patient.cart.remove_appointment(...) here